Tests/UnitTests/ConsentPreferencesManagerTests.swift

```diff
@@ -182,19 +299,34 @@ class ConsentPreferencesManagerTests: XCTestCase {
         ]
         let preferences2 = ConsentPreferences(consents: AnyCodable.from(dictionary: consents2)!)
 
-        // test pt. 2
+        // Test pt. 2
         XCTAssertTrue(manager.updateDefaults(with: preferences2))
-
-        // verify pt. 2
-        let flatDefaultConsents2 = manager.defaultPreferences?.asDictionary()?.flattening()
-
-        XCTAssertNil(flatDefaultConsents2?["consents.adID.val"] as? String)
-        XCTAssertEqual(flatDefaultConsents2?["consents.collect.val"] as? String, "y")
-        XCTAssertNotNil(flatDefaultConsents2?["consents.metadata.time"] as? String)
+        
+        // Verify pt. 2
+        let defaultConsents_pt2 = manager.defaultPreferences?.asDictionary()
+        
+        let expectedConsentsJSON_pt2 = #"""
+        {
+          "consents": {
+            "collect": {
+              "val": "y"
+            },
+            "metadata": {
+              "time": "STRING_TYPE"
+            }
+          }
+        }
+        """#
+        
+        // Verify default consents
+        assertExactMatch(
+            expected: getAnyCodable(expectedConsentsJSON_pt2)!,
+            actual: AnyCodable(AnyCodable.from(dictionary: defaultConsents_pt2)),
+            pathOptions: ValueTypeMatch(paths: "consents.metadata.time"), CollectionEqualCount(paths: nil))
```

Comment:
this is slightly confusing. can we have paths with default value nil for the CollectionEqualCount, reading CollectionEqualCount().

Also when would you use CollectionEqualCount with isActive false? is that equivalent to omitting it from the pathOptions?

Reply:
- I like the idea of using CollectionEqualCount or assertEqual even though some of the tests were not doing it already, in case extra keys are added unexpectedly it should fail.
- Also can we have KeyMustBeAbsent(paths: "consents.adID.val") to be consistent with the other APIs too.